File: frontend/src/lib/api/google-calendar.ts
import { apiDelete, apiGet, apiPost } from "@/lib/api";

export interface GoogleCalendarStatus {
  configured: boolean;
  connected: boolean;
  google_email: string | null;
  calendar_id: string | null;
  connected_at: string | null;
}

export const googleCalendarApi = {
  getStatus: () => apiGet<GoogleCalendarStatus>("/api/v1/integrations/google-calendar/status"),

  authorize: (returnUrl: string) =>
    apiPost<{ authorization_url: string }>("/api/v1/integrations/google-calendar/authorize", {
      return_url: returnUrl,
    }),

  disconnect: () => apiDelete("/api/v1/integrations/google-calendar"),
};

File: backend/app/schemas/google_calendar.py
"""Google Calendar integration API schemas."""

from datetime import datetime

from pydantic import BaseModel, EmailStr


class GoogleCalendarStatus(BaseModel):
    configured: bool
    connected: bool
    google_email: EmailStr | None = None
    calendar_id: str | None = None
    connected_at: datetime | None = None


class GoogleCalendarAuthorizeRequest(BaseModel):
    return_url: str | None = None


class GoogleCalendarAuthorizeResponse(BaseModel):
    authorization_url: str
